src/ddmd/traits.d

```diff
@@ -374,12 +374,16 @@ extern (C++) Expression semanticTraits(TraitsExp e, Scope* sc)
         printf("TraitsExp::semantic() %s\n", e.toChars());
     }
 
+    Scope* scx = sc.push();
```

Comment:
i've simply moved a block you've added this fall when there was the regression.

> Also not a fan of unspecifically removing visibility for all traits, can you please list which ones should behave like that and why?

The protection is bypassed for everything excepted `__traits(compiles)`. All the other traits are unbridled. The reason for that is that the user decides to respect or not the protection with `__traits(getProtection)`.